airflow/models/taskinstance.py

```diff
@@ -2144,7 +2145,11 @@ def _record_task_map_for_downstreams(self, value: Any, *, session: Session) -> N
             return
         if not isinstance(value, collections.abc.Collection) or isinstance(value, (bytes, str)):
             self.log.info("Failing %s for unmappable XCom push %r", self.key, type(value).__qualname__)
-            raise UnmappableXComPushed(value)
+            raise UnmappableXComTypePushed(value)
+        max_map_size = conf.getint("core", "max_map_size", fallback=128)
+        if len(value) > max_map_size:
+            self.log.info("Failing %s for oversize XCom push (%d > %d)", self.key, len(value), max_map_size)
```

Comment:
Do we need the log and the exception? Isn't the exception going to get logged elsewhere anyway?

Reply:
Merged.